### PR TITLE
Handle vCenter transport/ports when passed in install-config

### PR DESCRIPTION
**Pull request by SujanaN08**
Added port to install config platform datatype for vsphere.
Added port key to vshpere secret

### CODE REVIEW

File: pkg/asset/manifests/openshift.go [hunk 1]
```diff
@@ -157,6 +157,7 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 		cloudCreds = cloudCredsSecretData{
 			VSphere: &VSphereCredsSecretData{
 				VCenter:              installConfig.Config.VSphere.VCenter,
+				Port:                 installConfig.Config.VSphere.Port,
```
Comment (patrickdillon) [inline pkg/asset/manifests/openshift.go:160]: AFAIK we do not need to add port to the credentials. We need to add port to the `global` and `workspace` sections of the cloud provider config: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/vsphere/cloudproviderconfig.go

File: pkg/asset/manifests/template.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ type OpenStackCredsSecretData struct {
 // VSphereCredsSecretData holds encoded credentials and is used to generated cloud-creds secret
 type VSphereCredsSecretData struct {
 	VCenter              string
+	Port 				 string
```
Comment (patrickdillon) [inline pkg/asset/manifests/template.go:36]: AFAIK we do not need to add port to the credentials. We need to add port to the `global` and `workspace` sections of the cloud provider config: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/vsphere/cloudproviderconfig.go

File: pkg/types/vsphere/platform.go [hunk 3]
```diff
@@ -5,6 +5,9 @@ type Platform struct {
 	// VCenter is the domain name or IP address of the vCenter.
 	VCenter string `json:"vCenter"`
 
+	// Port is the transportation port for vcenter server
```
Comment (patrickdillon) [inline pkg/types/vsphere/platform.go:8]: nit: comments should be complete sentences and end with `.` or proper punctuation.

This is an optional field (`omitempty`) and should have the `\\ +optional` tag, similar to the `APIVIP` field. There are some other optional fields here as well, and it would be great if you could update those with the tag. We recently added support to [generate docs based on these comments](https://github.com/openshift/installer/blob/master/docs/dev/explain.md#generating-the-documentation). 

Any changes to these types should regenerate the docs, and it would be great to update the older fields so that the docs are properly generated.

File: pkg/types/vsphere/platform.go [hunk 4]
```diff
@@ -5,6 +5,9 @@ type Platform struct {
...
+	Port string `json:"port,omitempty"`
```
Comment (abhinavdahiya) [inline pkg/types/vsphere/platform.go:9]: Personally I would like that we allow user to give the vCenter location including the port. Having a separate field for port is not ergonomic.

Comment (patrickdillon) [reply pkg/types/vsphere/platform.go:9]: @abhinavdahiya vsphere cloud config has a separate field for port, so if they are combined we will need to `split`. Does that change your mind? It changed mine, but it is not a strong preference.

Comment (abhinavdahiya) [reply pkg/types/vsphere/platform.go:9]: Not really, the entire goal for our API is to make it easy for users to create the cluster, and we should take on complexity whenever it helps remove it for users..

File: pkg/asset/cluster/tfvars.go [hunk 5]
```diff
@@ -490,10 +490,15 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 
 		// Set this flag to use an existing folder specified in the install-config. Otherwise, create one.
 		preexistingFolder := installConfig.Config.Platform.VSphere.Folder != ""
-
+		vCenterURL := strings.SplitAfter(installConfig.Config.VSphere.VCenter, ":")
+		port := ""
+		if len(vCenterURL) > 1 && vCenterURL[1] != "443" {
+			port = vCenterURL[1]
+		}
```
Comment (patrickdillon) [inline pkg/asset/cluster/tfvars.go:497]: If `VSphere.VCenter` includes the port, which is now the working assumption, we no longer need to edit tfvars. When port was listed as a separate field, we needed to include port because ultimately we want to pass <VCenter>:<port> to tf. But since we are now including port in VCenter, no changes are necessary.

My suggestion about handling default port was also given with the first context and was really just a bit of a sugar. I would not worry about handling default port at all. If users provide default port along with their vcenter I would just treat it like any other valid port.

File: pkg/types/vsphere/validation/platform_test.go [hunk 6]
```diff
@@ -38,6 +38,15 @@ func TestValidatePlatform(t *testing.T) {
 			}(),
 			expectedError: `^test-path\.vCenter: Required value: must specify the name of the vCenter$`,
 		},
+		{
```
Comment (SujanaN08) [inline pkg/types/vsphere/validation/platform_test.go:41]: Work in progress

File: pkg/asset/installconfig/vsphere/validation.go [hunk 7]
```diff
@@ -28,6 +28,7 @@ func Validate(ic *types.InstallConfig) error {
 // ValidateForProvisioning performs platform validation specifically for installer-
 // provisioned infrastructure. In this case, self-hosted networking is a requirement
 // when the installer creates infrastructure for vSphere clusters.
+// when the installer creates infrastructure for vSphere clusters.
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/validation.go:31]: duplicate line

File: pkg/types/vsphere/validation/platform.go [hunk 8]
```diff
@@ -94,3 +100,21 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
...
+// ValidateClient validates vCenter to only contain hostname and port
+func ValidateClient(vCenter string) error {
+	vCenterURL, err := url.Parse(vCenter)
+	if err != nil {
+		return err
+	}
+	if vCenterURL.Scheme != "" {
+		return errors.New("vCenter hostname should not contain url scheme")
+	}
+	if vCenterURL.Hostname() == "" {
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:113]: Did you test this out? It doesn't seem like `Hostname` is going to work: https://play.golang.org/p/0XjNkuIdP52

Abhinav's suggestion of https://golang.org/pkg/net/#SplitHostPort seems like a better fit.

File: pkg/types/vsphere/validation/platform.go [hunk 9]
```diff
@@ -94,3 +100,21 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateClient validates vCenter to only contain hostname and port
+func ValidateClient(vCenter string) error {
```
Comment (abhinavdahiya) [inline pkg/types/vsphere/validation/platform.go:105]: i think this needs a better name, like ValidateVCenterAddress or something like that...

File: pkg/asset/installconfig/vsphere/vsphere.go [hunk 10]
```diff
@@ -103,7 +104,10 @@ func getClients() (*vCenterClient, error) {
 				Message: "vCenter",
 				Help:    "The hostname of the vCenter to be used for installation.",
 			},
-			Validate: survey.Required,
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				err := validation.ValidateClient(ans.(string))
+				return err
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/vsphere/vsphere.go:109]:
```suggestion
				return validation.ValidateClient(ans.(string))
```

File: pkg/asset/manifests/openshift.go [hunk 11]
```diff
@@ -156,7 +157,7 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 	case vspheretypes.Name:
 		cloudCreds = cloudCredsSecretData{
 			VSphere: &VSphereCredsSecretData{
-				VCenter:              installConfig.Config.VSphere.VCenter,
+				VCenter:              url.URL.Port(installConfig.Config.VSphere.VCenter),
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/openshift.go:160]: I don't understand what you are trying to do here ??

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 12]
```diff
@@ -26,7 +28,10 @@ func CloudProviderConfig(folderPath string, p *vspheretypes.Platform) (string, e
 	fmt.Fprintln(buf, "")
 
 	fmt.Fprintln(buf, "[Workspace]")
-	printIfNotEmpty(buf, "server", p.VCenter)
+	if port := url.URL.Port(p.VCenter); port != "" {
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:31]: I don't understand what you are trying to do here ??

File: pkg/asset/manifests/openshift.go (hunk 11 above)
Comment (abhinavdahiya) [reply pkg/asset/manifests/openshift.go:160]: So you want to get just the hostname.. ?

you probably need to 

```
u, err := url.Parse(installConfig.Config.VSphere.VCenter)
// handle err
u.Hostname() or u.Port()
```

File: pkg/types/vsphere/validation/platform.go [hunk 13]
```diff
@@ -14,6 +16,10 @@ import (
 func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if len(p.VCenter) == 0 {
+		vCenter := p.VCenter
```
Comment (abhinavdahiya) [inline pkg/types/vsphere/validation/platform.go:19]: why the initialization?

Comment (abhinavdahiya) [reply pkg/types/vsphere/validation/platform.go:19]: also you are testing this in an if block that is only true when vCenter is zero... so i think this one needs to outside the if block.

**Comment by patrickdillon** [comment]
I think my suggestion of URL library misled you. https://golang.org/pkg/net/#SplitHostPort definitely seems like a better fit. Also you might consider helper functions in the platform to get hostname and port

so you could do something like:
`installConfig.Config.VSphere.VCenterPort()` 
`installConfig.Config.VSphere.VCenterHostname()` 
which would make use of the `SplitHostPort` call. Just a suggestion.

File: pkg/asset/cluster/tfvars.go [hunk 14]
```diff
@@ -490,7 +490,6 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 
 		// Set this flag to use an existing folder specified in the install-config. Otherwise, create one.
 		preexistingFolder := installConfig.Config.Platform.VSphere.Folder != ""
-
```
Comment (patrickdillon) [inline pkg/asset/cluster/tfvars.go:493]: no reason to keep this line deletion in this pr

File: pkg/asset/manifests/openshift.go [hunk 15]
```diff
@@ -154,9 +155,16 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 			},
 		}
 	case vspheretypes.Name:
+		vCenterURL, err := url.Parse(installConfig.Config.VSphere.VCenter)
+		host := installConfig.Config.VSphere.VCenter
+		if err == nil {
+			if hostname := vCenterURL.Hostname(); hostname != "" {
+				host = hostname
+			}
+		}
```
Comment (patrickdillon) [inline pkg/asset/manifests/openshift.go:164]: this also is not working for the same reasons I pointed out in the other comment: https://github.com/openshift/installer/pull/3836#discussion_r451723695

you can test this like this:

1. create an install config which includes a port 
```console
$ cat install-config.yaml | grep vCenter
    vCenter: vcsa.vmware.devcluster.openshift.com:6443
```
2 create manifests
```
$ openshift-install create manifests
```

3 check the manifest

```
$ cat openshift/99_cloud-creds-secret.yaml 
kind: Secret
apiVersion: v1
metadata:
  namespace: kube-system
  name: vsphere-creds
data:
  vcsa.vmware.devcluster.openshift.com:6443.username: <redacted>
  vcsa.vmware.devcluster.openshift.com:6443.password: <redacted>

```

the goal here is that `vcsa.vmware.devcluster.openshift.com:6443.username` would be `vcsa.vmware.devcluster.openshift.com.username`

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 16]
```diff
@@ -3,8 +3,8 @@ package vsphere
 import (
 	"bytes"
 	"fmt"
-
 	vspheretypes "github.com/openshift/installer/pkg/types/vsphere"
+	"net/url"
```
Comment (patrickdillon) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:7]: imports should be organized like this:

```
<stdlib>

<external>

<internal(installer)>
```

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 17]
```diff
@@ -16,6 +16,8 @@ func printIfNotEmpty(buf *bytes.Buffer, k, v string) {
 // CloudProviderConfig generates the cloud provider config for the vSphere platform.
 // folderPath is the absolute path to the VM folder that will be used for installation.
 // p is the vSphere platform struct.
+// We check to see if a non-default port value is specified for vcenter("url:port") and send them as
```
Comment (patrickdillon) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:19]: nit: we are checking for any ports, including default

File: pkg/types/vsphere/validation/platform.go [hunk 18]
```diff
@@ -2,6 +2,8 @@ package validation
 
 import (
 	"fmt"
+	"github.com/pkg/errors"
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:5]: this lib should be moved to the next block of imports

File: pkg/types/vsphere/validation/platform.go [hunk 19]
```diff
@@ -94,3 +100,18 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:108]: I don't think this check is necessary. vCenter is a required field so it should be checked before this. try removing and see what happens.

File: pkg/types/vsphere/validation/platform.go [hunk 20]
```diff
@@ -94,3 +100,18 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:104]: nit: comments should end with a period

File: pkg/types/vsphere/validation/platform.go [hunk 21]
```diff
@@ -94,3 +100,18 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
...
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
+	vCenterURL, err := url.Parse(vCenter)
+	if err != nil {
+		return err
+	}
+	if vCenterURL.RawQuery != "" {
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:113]: we need to validate that this does not contain a transport, like https://

File: pkg/types/vsphere/validation/platform_test.go [hunk 22]
```diff
@@ -38,6 +38,15 @@ func TestValidatePlatform(t *testing.T) {
...
+			name: "invalid vcenter name",
+			platform: func() *vsphere.Platform {
+				p := validPlatform()
+				p.VCenter = "http://forest.com:9090?param=value"
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform_test.go:45]: this test case is failing because it includes params. it would be better to include the actual error message so that is clear, and also to include another test case that fails for transport

File: pkg/types/vsphere/validation/platform.go (hunk 19 above)
Comment (SujanaN08) [reply pkg/types/vsphere/validation/platform.go:108]: When I give something like "       " in vCenter name it is read a string of spaces and fails. I have added a validation.

File: pkg/types/vsphere/validation/platform.go (hunk 21 above)
Comment (SujanaN08) [reply pkg/types/vsphere/validation/platform.go:114]: SplitHostPort just splits a string based on the occurence of first ":". url.Parse(urlString) parses a string and separates transport scheme from domain based on the first occurence of ":".

File: pkg/types/vsphere/validation/platform_test.go (hunk 22 above)
Comment (SujanaN08) [reply pkg/types/vsphere/validation/platform_test.go]: I tried different regex strings. I had no luck with the test accepting them. The actual error message has full message which says there should not be params/scheme in vCenter. For the test I shortened it.

**Comment by SujanaN08** [comment]
Parse(url string) and ParseRequestURI(url string) methods are not sufficient to correctly validate vCenter being passed in install congif. 
Both these methods consider the string before the first occurence of a ":" as transport scheme for a url. 

- For "hostname:port" a url parser recognizes "hostname" as Scheme and "port" as hostname. 

- For "https://hostname:port" a url parser recognizes "https" as Scheme and "hostname" as Host and "port" as Port.

This results in inconsistency and therby we cannot use the parser as a validation.

SplitHostPort method from net package splits any string based on the occrence of ":". If there is more than one ":", eg:  "https://hostname:port" the methods throws an error. Therefore if we have a vCenter name "https://hostname" it returns "https" as host and rest of the string as port. Therefore we cannot use SplitHostPort effectively as a means to validate vCenter.

The best method I thought is to use 

- url parser to detect request params. 

- make sure there is only one occurence of ":"

- split host and port with the help of SplitHostPort method.

- Validate port to have only numbers

- Validate host to have only alphanumeric values and . , -

**Comment by patrickdillon** [comment]
Reiterating our in-person chat: we do have to check that "https://" is not included in the vCenter. This is common for users to do, and we want to catch it.

To me it seems like the best approach is to use `net/url` for validation and `net.SplitHostPort()` for the rest.

It looks like in `net/url` we should be using `ParseRequestURI` instead of `Parse` (I found this through [this stack overflow answer](https://stackoverflow.com/questions/31480710/validate-url-with-standard-package-in-go), in case that helps). The source code has a really good explanation here: https://github.com/golang/go/blob/master/src/net/url/url.go#L340-L369

Using `ParseRequestURI` you should be able to check that `vCenter` does not contain hostname, and,  any provided port is valid (one way you could do this is manually add "https://" only when checking for port & hostname and see if validation fails).

File: pkg/types/vsphere/validation/platform.go [hunk 23]
```diff
@@ -94,3 +99,42 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port.
+func ValidateVCenterAddress(vCenter string) error {
+	if strings.TrimSpace(vCenter) == "" {
+		return errors.New("must specify the name of the vCenter")
+	}
+	vCenterURL, err := url.ParseRequestURI(vCenter)
```
Comment (SujanaN08) [inline pkg/types/vsphere/validation/platform.go:108]: I have used ParseRequestURI() to check for url scheme. 
It does not return an error if there is a scheme present in the url string.
eg: https:forest.com
https://forest.com 
Both the above examples would not throw an error. 
"https:forest.com" returns an empty host and "https://forest.com" returns "forest.com" as host. 

In the later part of the method I am using url.Parse() to check for params and path because url.ParseRequestURI() would throw an error if a scheme is not mentioned and would possibly make it difficult to test other things.

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 24]
```diff
@@ -26,13 +28,18 @@ func CloudProviderConfig(folderPath string, p *vspheretypes.Platform) (string, e
 	fmt.Fprintln(buf, "")
 
 	fmt.Fprintln(buf, "[Workspace]")
-	printIfNotEmpty(buf, "server", p.VCenter)
+	host, port, err := net.SplitHostPort(p.VCenter)
+	if err != nil {
+		host = p.VCenter
+	}
+	printIfNotEmpty(buf, "port", port)
```
Comment (abhinavdahiya) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:35]: we should not be setting the port when the vCenter didn't have that in the first place right? otherwise we would set it to the zero value (0)

File: pkg/types/validation/installconfig_test.go [hunk 25]
```diff
@@ -665,7 +665,7 @@ func TestValidateInstallConfig(t *testing.T) {
 				c.Platform.VSphere.VCenter = ""
 				return c
 			}(),
-			expectedError: `^platform\.vsphere.vCenter: Required value: must specify the name of the vCenter$`,
+			expectedError: `^platform\.vsphere.vCenter: Invalid value: "": must specify the name of the vCenter$`,
```
Comment (abhinavdahiya) [inline pkg/types/validation/installconfig_test.go:668]: i think like the previous error better where we say it is required instead of saying `"" is invalid value`

File: pkg/types/vsphere/validation/platform.go [hunk 26]
```diff
@@ -94,3 +99,42 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateVCenterAddress validates vCenter to only contain hostname and port.
+func ValidateVCenterAddress(vCenter string) error {
```
Comment (abhinavdahiya) [inline pkg/types/vsphere/validation/platform.go:104]: can we add unit tests for this specific function...

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go (hunk 24 above)
Comment (SujanaN08) [reply pkg/asset/manifests/vsphere/cloudproviderconfig.go:35]: printIfNotEmpty will only set if there is a non empty string in port value. Port is a string so ""(empty string) will not be considered.

File: pkg/types/validation/installconfig_test.go (hunk 25 above)
Comment (SujanaN08) [reply pkg/types/validation/installconfig_test.go:668]: I am using custom validation where "" or "     " or "\n" or "\t" is checked and reported as error. I am not using the defaut valitation, where we just check if the string is of 0 length, at all.

File: pkg/types/vsphere/validation/platform.go [hunk 27]
```diff
@@ -94,3 +99,41 @@ func validateFolder(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
...
+	}
+	vCenterURL, err := url.ParseRequestURI(vCenter)
+	if err == nil && vCenterURL.Host != "" {
+		return errors.Errorf("vCenter hostname cannot contain url scheme")
+	}
+
+	vCenterURL, err = url.Parse(vCenter)
+	if err != nil {
+		return errors.Errorf("vCenter hostname is not valid")
+	}
```
Comment (patrickdillon) [inline pkg/types/vsphere/validation/platform.go:116]: I'm not sure I understand why you are doing this (checking for hostname validity twice--you check again below). I know I suggested something like this, so I'm not sure if my suggestion was unclear. This is not what I had in mind. I was suggesting that INSTEAD of the manual checking host and port, you could leverage `url.Parse` or `url.ParseRequestURI`. 

The entire flow would look something like this:
1. Check if vCenter contains a scheme if it does return an error (you are already doing this step)
2. We know vCenter does not contain a scheme, so create a temporary variable and prepend the scheme to vcenter
3. Run that temporary variable through one of the parse methods, which will then check for correct domain and port
4. if there is an error make sure to replace the temporary variable with the actual value passed

I have created a playground to illustrate how parse checks the domain and port: https://play.golang.org/p/S_CutN804xt

You are free to use this idea--and it could potentially be improved so that the checks in step 3 are run even if the step 1 check fails, but we could discuss that later.

The other checks you have already included are also fine, but I'm not sure why this `Parse` is here.

Comment (SujanaN08) [reply pkg/types/vsphere/validation/platform.go:116]: Yeah your approach is more understandable. I will modify it.

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 28]
```diff
@@ -3,8 +3,8 @@ package vsphere
 import (
 	"bytes"
 	"fmt"
-
 	vspheretypes "github.com/openshift/installer/pkg/types/vsphere"
+	"net"
```
Comment (patrickdillon) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:7]: github won't let me make a suggestion due to the deleted line. It should look like this:
```go
"bytes"
"fmt"
"net"

vspheretypes "github.com/openshift/installer/pkg/types/vsphere"
```

File: pkg/asset/manifests/vsphere/cloudproviderconfig.go [hunk 29]
```diff
@@ -16,6 +16,8 @@ func printIfNotEmpty(buf *bytes.Buffer, k, v string) {
 // CloudProviderConfig generates the cloud provider config for the vSphere platform.
 // folderPath is the absolute path to the VM folder that will be used for installation.
 // p is the vSphere platform struct.
+// We check to see if a port value is specified for vcenter("url:port") and send them as
+// separate fields in cloud config.
```
Comment (patrickdillon) [inline pkg/asset/manifests/vsphere/cloudproviderconfig.go:20]: nit: I'm not sure if this comment belongs here for the function-level documentation. also it probably would belong before the list/description of parameters.

**Comment by SujanaN08** [comment]
Currently only masters are being created and I think adding the host and port to machine config would let the worker nodes to be created successfully. @jhixson74
